src/plutus-core-execution.k

```diff
@@ -1,99 +1,87 @@
-    when F1 <Float F2
-  rule ( builtin lessThanFloat:Name ( F1:Float F2:Float ) ) => #primFalse
-    when notBool (F1 <Float F2)
-  rule ( builtin equalsFloat:Name   ( F1:Float F2:Float ) ) => #primTrue
-    when F1 ==Float F2
-  rule ( builtin equalsFloat:Name   ( F1:Float F2:Float ) ) => #primFalse
-    when F1 =/=Float F2
-
-  syntax String ::= #toHex(Int)                                      [function]
-                  | #hex(Int)                                        [function]
-
-  rule #hex(0)  => "0"
-  rule #hex(1)  => "1"
-  rule #hex(2)  => "2"
-  rule #hex(3)  => "3"
-  rule #hex(4)  => "4"
-  rule #hex(5)  => "5"
-  rule #hex(6)  => "6"
-  rule #hex(7)  => "7"
-  rule #hex(8)  => "8"
-  rule #hex(9)  => "9"
-  rule #hex(10) => "A"
-  rule #hex(11) => "B"
-  rule #hex(12) => "C"
-  rule #hex(13) => "D"
-  rule #hex(14) => "E"
-  rule #hex(15) => "F"
-
-  // This will fail on 0, so should not be used by itself. See the rules for
-  // `(builtin intToByteString (...))`.
-  rule #toHex(0) => ""
-  rule #toHex(N:Int) => #toHex(N /Int 16) +String #hex(N %Int 16)
-    when N =/=Int 0
-
-  rule ( builtin ceil:Name  ( F:Float ) ) => #tmHolder(ceilFloat(F))
-  rule ( builtin floor:Name ( F:Float ) ) => #tmHolder(floorFloat(F))
-  rule #tmHolder(F:Float) => F
-  rule ( builtin round:Name ( F:Float ) )
-    => Float2Int(roundFloat(F:Float, 2, 2))
+  rule ( builtin addFloat:Name      F1:PlFloat F2:PlFloat )
+    => #toPlFloat(#toFloat(F1) +Float #toFloat(F2))
+  rule ( builtin subtractFloat:Name F1:PlFloat F2:PlFloat )
+    => #toPlFloat(#toFloat(F1) -Float #toFloat(F2))
+  rule ( builtin multiplyFloat:Name F1:PlFloat F2:PlFloat )
+    => #toPlFloat(#toFloat(F1) *Float #toFloat(F2))
+  rule ( builtin divideFloat:Name   F1:PlFloat F2:PlFloat )
+    => #toPlFloat(#toFloat(F1) /Float #toFloat(F2))
+  rule ( builtin lessThanFloat:Name F1:PlFloat F2:PlFloat )
+    => #primTrue when #toFloat(F1) <Float #toFloat(F2)
+  rule ( builtin lessThanFloat:Name F1:PlFloat F2:PlFloat )
+    => #primFalse when notBool (#toFloat(F1) <Float #toFloat(F2))
+  rule ( builtin equalsFloat:Name   F1:PlFloat F2:PlFloat )
+    => #primTrue when #toFloat(F1) ==Float #toFloat(F2)
+  rule ( builtin equalsFloat:Name   F1:PlFloat F2:PlFloat )
+    => #primFalse when #toFloat(F1) =/=Float #toFloat(F2)
+
+
+  rule ( builtin ceil:Name  F:PlFloat ) => #toPlFloat(ceilFloat(#toFloat(F)))
+  rule ( builtin floor:Name F:PlFloat ) => #toPlFloat(floorFloat(#toFloat(F)))
+  rule ( builtin round:Name F:PlFloat )
+    => Float2Int(roundFloat(#toFloat(F), 2, 2))
+
+  syntax PlFloat ::= #toPlFloat(Float)                                [function]
+  rule #toPlFloat(F:Float)
+    => #parseToken("PlFloat@PLUTUS-CORE-SYNTAX", Float2String(F))
+  rule #toFloat(F:PlFloat)
+    => #parseToken("Float@FLOAT-SYNTAX", #toStringFloat(F))
 
   syntax ByStr ::= toByStrVerbatim ( String )                        [function]
   rule toByStr(S:String)
```

Comment:
`toByStrVerbatim` is declared but never used, while `toByStr` is used but never declared

Also, these should begin with a `#`